heat/decomposition/tests/test_pca.py

```diff
@@ -210,3 +209,122 @@ def test_pca_randomized(self):
 
         pca = ht.decomposition.PCA(n_components=None, svd_solver="randomized", random_state=1234)
         self.assertEqual(ht.random.get_state()[1], 1234)
+
+
+class TestIncrementalPCA(TestCase):
+    def test_incrementalpca_setup(self):
+        pca = ht.decomposition.IncrementalPCA(n_components=2)
+
+        # check correct base classes
+        self.assertTrue(ht.is_estimator(pca))
+        self.assertTrue(ht.is_transformer(pca))
+
+        # check correct default values
+        self.assertEqual(pca.n_components, 2)
+        self.assertEqual(pca.whiten, False)
+        self.assertEqual(pca.batch_size, None)
+        self.assertEqual(pca.components_, None)
+        self.assertEqual(pca.singular_values_, None)
+        self.assertEqual(pca.mean_, None)
+        self.assertEqual(pca.n_components_, None)
+        self.assertEqual(pca.batch_size_, None)
+        self.assertEqual(pca.n_samples_seen_, 0)
+
+        # check catching of invalid parameters
+        # whitening and in-place are not yet supported
+        with self.assertRaises(NotImplementedError):
+            ht.decomposition.IncrementalPCA(whiten=True)
+        with self.assertRaises(NotImplementedError):
+            ht.decomposition.IncrementalPCA(copy=False)
+        # wrong n_components
+        with self.assertRaises(TypeError):
+            ht.decomposition.IncrementalPCA(n_components=0.9)
+        with self.assertRaises(ValueError):
+            ht.decomposition.IncrementalPCA(n_components=0)
+
+    def test_incrementalpca_part1(self):
```

Comment:
Use more descriptive names and split test into multiple categories. Part 1 could be `test_incrementalpca_full_rank`, and part 2 could be 'test_incremenatlpca_truncation happens'

Reply:
renaming done, introduced another category for catching wrong inputs